src/highlight/highlight.ts

```diff
@@ -1,59 +1,35 @@
-import { Component, Input } from '@angular/core';
-import { bem, getPropertyByPath } from '../utils';
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+
+import { Hit } from 'instantsearch.js';
+
+import { highlight } from 'instantsearch.js/es/helpers';
+import { getPropertyByPath } from 'instantsearch.js/es/lib/utils';
 
 @Component({
   selector: 'ais-highlight',
-  template: `<span [class]="cx()" [innerHtml]="content"></span>`,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `<span class="ais-Highlight" [innerHtml]="content"></span>`,
 })
 export class NgAisHighlight {
   @Input() attribute: string;
-  @Input() hit: { _highlightResult?: {}; label?: string; highlighted?: string };
-  @Input() tagName: string = 'em';
-
-  cx = bem('Highlight');
+  @Input() hit: Partial<Hit>;
+  @Input() tagName: string = 'mark';
```

Comment:
That's a breaking change so we may want to keep `em` for now?

Reply:
In the current released version of Angular InstantSearch highlighted text is already between `mark` elements, making the addition of the `ais-Highlight-highlighted` class name impossible. This fixes that issue.

Reply:
[Looks like `mark` is being used indeed](https://codesandbox.io/s/github/algolia/doc-code-samples/tree/master/Angular%20InstantSearch/getting-started). Are the tests unreliable?

Reply:
The tests fixtures weren't correct indeed. I gave more details on why the current implementation would not work here: https://github.com/algolia/angular-instantsearch/pull/888#discussion_r791860540.